packages/@dcl/inspector/src/components/Tree/Tree.tsx

```diff
@@ -212,13 +268,32 @@ export function Tree<T>() {
 
       return (
         <div className={treeClassNames} data-test-id={id} data-test-label={label}>
-          <div style={getLevelStyles(level)} className={cx({ selected, item: true, hidden })}>
+          <div
+            style={getLevelStyles(level)}
+            className={cx({ selected, item: true, hidden })}
+            onMouseEnter={handleOnHover}
+            onMouseLeave={handleOnHover}
+          >
             <ContextMenu {...controlsProps} />
             <div ref={ref} style={getEditModeStyles(editMode)} className="item-area">
               <DisclosureWidget enabled={enableOpen} isOpen={open} onOpen={handleOpen} />
               <div onClick={handleSelect} onContextMenu={handleSelect} className="selectable-area">
                 {props.getIcon ? props.getIcon(value) : <></>}
                 <div>{label || id}</div>
+                {isEntity && (
+                  <div className={cx('action-area', { 'item-locked': isEntityLocked, 'item-hidden': isEntityHidden })}>
+                    {isEntityLocked ? (
+                      <LockIcon className="lock-icon" size={16} onClick={handleToggleLockComponent} />
+                    ) : (
+                      <UnlockIcon className="unlock-icon" size={16} onClick={handleToggleLockComponent} />
+                    )}
+                    {isEntityHidden ? (
+                      <InvisibleIcon className="invisible-icon" size={16} onClick={handleToggleHideComponent} />
+                    ) : (
+                      <VisibleIcon className="visible-icon" size={16} onClick={handleToggleHideComponent} />
+                    )}
+                  </div>
```

Comment:
maybe we can move some of this to another component? so we avoid adding more complexity to this one, which is already pretty hard to follow along

Reply:
sure! doing it